src/wmtk/io/EdgeMeshReader.cpp

```diff
@@ -0,0 +1,86 @@
+#include "EdgeMeshReader.hpp"
+#include <spdlog/spdlog.h>
+
+namespace wmtk {
+EdgeMeshReader::EdgeMeshReader(const std::string& filename, const file_type type)
+    : m_filename(filename)
+    , m_type(type)
+{}
+void EdgeMeshReader::read(Eigen::Matrix<long, -1, -1>& E, Eigen::MatrixXd& V)
+{
+    std::vector<std::pair<long, long>> edges;
+    std::vector<std::vector<double>> vertices;
+    char* line_buffer = nullptr;
+    std::fstream f(m_filename);
+    if (!f) {
```

Comment:
`f.is_open()` might do the same thing (I am not sure) but it for sure makes your intend more clear.